cmd/kube-scheduler/app/server.go

```diff
@@ -541,20 +545,26 @@ func createClients(config componentconfig.ClientConnectionConfiguration, masterO
 
 	client, err := clientset.NewForConfig(restclient.AddUserAgent(kubeConfig, "scheduler"))
 	if err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, nil, err
 	}
 
 	leaderElectionClient, err := clientset.NewForConfig(restclient.AddUserAgent(kubeConfig, "leader-election"))
 	if err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, nil, err
 	}
 
 	eventClient, err := clientset.NewForConfig(kubeConfig)
 	if err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, nil, err
+	}
+	var metricsClient *resourceclient.MetricsV1beta1Client
+	if utilfeature.DefaultFeatureGate.Enabled(features.UsageBasedScheduling) {
+		metricsClient, err = resourceclient.NewForConfig(kubeConfig)
```

Comment:
How this error will be handled?

Actually this raised the question: what if user does not have metric api server installed? Should we fail the scheduler or  just log error and disable `UsageBasedScheduling`?

Reply:
Good question. I thought about and my initial inclination was same as yours but while thinking more about it, I realized it may cause inconsistent user behaviour. For example,
- Say enduser set this flag.
-  In the codepath, if we reset the flag here when we see an error, say after sometime kube-metrics server has been deployed, we won't see error. Now, this part of code won't be run again(as this is initializing part) and usage based scheduling will be off, even though user thinks that he/she has set the flag causing inconsistent user behavior.

But let me think, if there is a better way to handle it.